### DISCUSSION TITLE
pytask Permission denied

### DISCUSSION THREAD

**Discussion by zhouhoo** [apache/dolphinscheduler#11290] [Q&A]
hello, Had anyone encounter such problem when create and run a python workflow?

[LOG-PATH]: /opt/dolphinscheduler/logs/20220804/6419126763488_1-5484-16467.log, [HOST]:  Host{address='192.168.32.6:1234', ip='192.168.32.6', port=1234}
[INFO] 2022-08-04 04:04:47.018 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[83] - python task params {"localParams":[],"rawScript":"print(123456)","resourceList":[]}
[INFO] 2022-08-04 04:04:47.020 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[225] - raw python script : print(123456)
[INFO] 2022-08-04 04:04:47.020 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[178] - tenantCode :root, task dir:/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467
[INFO] 2022-08-04 04:04:47.021 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[181] - generate python script file:/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/py_5484_16467.py
[INFO] 2022-08-04 04:04:47.021 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[188] - #-*- encoding=utf8 -*-

print(123456)
[INFO] 2022-08-04 04:04:47.021 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[85] - tenantCode user:root, task dir:5484_16467
[INFO] 2022-08-04 04:04:47.021 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[90] - create command file:/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command
[INFO] 2022-08-04 04:04:47.021 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[116] - command : #!/bin/sh
BASEDIR=$(cd `dirname $0`; pwd)
cd $BASEDIR
source /opt/dolphinscheduler/conf/dolphinscheduler_env.sh
${PYTHON_HOME} /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/py_5484_16467.py
[INFO] 2022-08-04 04:04:47.025 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[290] - task run command: sudo -u root sh /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command
[INFO] 2022-08-04 04:04:47.026 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[181] - process start, process id is: 19988
[INFO] 2022-08-04 04:04:47.043 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[205] - process has exited, execute path:/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467, processId:19988 ,exitStatusCode:126 ,processWaitForStatus:true ,processExitValue:126
[INFO] 2022-08-04 04:04:48.027 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[63] -  -> /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: 4: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: source: not found
	/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: 5: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/py_5484_16467.py: Permission denied
[INFO] 2022-08-04 04:04:48.029 +0000 [taskAppId=TASK-20220804-6419126763488_1-5484-16467] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.python.PythonTask:[57] - FINALIZE_SESSION

the python file can not be created, only part of the filepath is create, eg: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/py_5484_16467.py

the  16467/py_5484_16467.py   do not exsist!!

**Comment by dyyTT** [comment]
Hello, I have encountered the same problem. Have you solved this problem.

**Reply by DarkAssassinator** [reply]
i think that 16467/py_5484_16467.py exsist, because it will be deleted after task finish. 
u should check the following error message:
/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: **source: not found**
/tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: 5: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/5484_16467.command: /tmp/dolphinscheduler/exec/process/5888838384480/6419126763488_1/5484/16467/py_5484_16467.py: **Permission denied**

**Comment by DarkAssassinator** [comment]
ohh, i think there is a issue, becuase DS use `#!/bin/sh`, not `#!/bin/bash`

**Reply by dyyTT** [reply]
![image](https://user-images.githubusercontent.com/50373480/187335116-cba6413c-9365-4973-bda8-66d29dfd8ce2.png)
However, after I modified the code and rerun it, the same problem still occurred

**Comment by dyyTT** [comment]
![image](https://user-images.githubusercontent.com/50373480/187363118-d54439ec-59d0-434e-aded-96885f1db895.png)
I modified the run command and the environment variable file, and then the operation succeeded.

**Reply by DarkAssassinator** [reply]
because no ${PYTHOM_HOME} ?

**Comment by dyyTT** [comment]
yes，I started it with docker compose，However, the "python_home" configured in the. Env file is not read，Therefore, I directly modified the "python_home" environment variable in the "Dolphin scheduler_env. Sh" file. After re build image,It works。
![image](https://user-images.githubusercontent.com/50373480/187573485-46de24ee-cf14-4a27-8ff0-1cc111cc25b9.png)
